### PR TITLE
Request UAC via SMS (HH, Ind & CE1)

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
New CE UAC fulfilment codes and new treatment ID sender and receiver codes need including 

# What has changed
New CE UAC fulfilment codes included in existing test
Updated notify template IDs

# How to test?
Build repos on card and run the ATs

# Links
https://trello.com/c/0zPdVGFQ/941-request-uac-via-sms-hh-ind-ce1-5
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SDC&title=UAC+by+SMS+Fulfilment+2021
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SDC&title=RM+Product+Library

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[Request UAC via SMS (HH, Ind & CE1) (5)](https://trello.com/c/0zPdVGFQ/941-request-uac-via-sms-hh-ind-ce1-5)

**Review by NickGrantX — CHANGES_REQUESTED** [review]
We need test coverage for CE/SPG individual SMS fulfillments (i.e. without an individual case ID)

File: acceptance_tests/features/steps/fulfilment.py [hunk 1]
```diff
@@ -187,6 +187,58 @@ def create_individual_print_fulfilment_message(context, fulfilment_code):
             routing_key=Config.RABBITMQ_FULFILMENT_REQUESTED_ROUTING_KEY)
 
 
+@step('an individual UAC SMS fulfilment request "{fulfilment_code}" is received by RM')
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/fulfilment.py]: I would make this step definition and function name clear that this is for CE/SPGs. HH has an indiv case ID

File: acceptance_tests/features/fulfilment.feature [hunk 2]
```diff
@@ -295,3 +305,10 @@ Feature: Handle fulfilment request events
     And a UAC updated message with "21" questionnaire type is emitted
     And print file and manifest files are created for pack code "P_OR_I1" with individual contact details
     Then the fulfilment request case has these events logged [SAMPLE_LOADED,FULFILMENT_REQUESTED,RM_UAC_CREATED,PRINT_CASE_SELECTED]
+
+  Scenario: Individual UAC SMS fulfilments without Individual Case ID
```
Comment (NickGrantX) [inline acceptance_tests/features/fulfilment.feature]: could say "CE/SPG" instead of "without indiv case ID"

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Wording changes to clarify test/function/scenario purpose

File: acceptance_tests/features/fulfilment.feature [hunk 3]
```diff
@@ -12,13 +12,23 @@ Feature: Handle fulfilment request events
...
       | UACHHT2         | 02                 | household Welsh and English |
       | UACHHT2W        | 03                 | household Welsh             |
       | UACHHT4         | 04                 | household Northern Ireland  |
 
+    @regression
+    Examples: CE UAC fulfilment codes: <fulfilment code>
+      | fulfilment code | questionnaire type | SMS template         |
+      | UACCET2         | 32                 | ce Welsh and English |
+      | UACCET2W        | 33                 | ce Welsh             |
+
```
Comment (AdamHawtin) [inline acceptance_tests/features/fulfilment.feature]: This scenario outline only ever loads and HH case, I think these example need to load different samples with the correct case type.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: acceptance_tests/features/fulfilment.feature [hunk 4]
```diff
@@ -295,3 +313,10 @@ Feature: Handle fulfilment request events
     And a UAC updated message with "21" questionnaire type is emitted
     And print file and manifest files are created for pack code "P_OR_I1" with individual contact details
     Then the fulfilment request case has these events logged [SAMPLE_LOADED,FULFILMENT_REQUESTED,RM_UAC_CREATED,PRINT_CASE_SELECTED]
+
+  Scenario: Individual UAC SMS fulfilments for a CE/SPG case
+    Given sample file "sample_1_english_SPG_estab.csv" is loaded successfully
+    When an individual UAC SMS fulfilment request "UACIT1" is received by RM for a CE/SPG case
+    And a UAC updated message with "21" questionnaire type is emitted
+    Then notify api was called with SMS template "UACIT1"
+    And the fulfilment request case has these events logged [SAMPLE_LOADED,FULFILMENT_REQUESTED,RM_UAC_CREATED]
```
Comment (AdamHawtin) [inline acceptance_tests/features/fulfilment.feature:322]: I think this scenario also needs to check the UAC is linked to the parent case, it's implied by the events but best to be specific

Comment (NickGrantX) [reply acceptance_tests/features/fulfilment.feature:322]: I think it does that already, by implication (i.e. there is no 'parent' case for this scenario)

Comment (AdamHawtin) [reply acceptance_tests/features/fulfilment.feature:322]: Fair enough, I suppose it's a very strong implication in that the scenario doesn't mention a HI/individual case

**Review by NickGrantX — APPROVED** [review]
ATs pass

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**